lib/modules/dosomething/dosomething_search/dosomething_search.module

```diff
@@ -181,6 +181,9 @@ function dosomething_search_filter_types($type) {
 function dosomething_search_apachesolr_index_document_build_node(ApacheSolrDocument $document, $entity, $env_id) {
   if ($entity->type == 'campaign' || $entity->type == 'campaign_group') {
     $wrapper = entity_metadata_wrapper('node', $entity);
+    global $language_content;
+    $wrapper->language($language_content->language);
```

Comment:
There's a helper function to do just that: `dosomething_helpers_node_metadata_wrapper()`.